src/test/regress/expected/failure_connection_establishment.out

```diff
@@ -314,6 +313,107 @@ SELECT * FROM citus_check_connection_to_node('localhost', :worker_2_proxy_port);
  f
 (1 row)
 
+-- tests for citus_check_cluster_node_health
+-- kill all connectivity checks that originate from this node
+SELECT citus.mitmproxy('conn.onQuery(query="^SELECT citus_check_connection_to_node").kill()');
+ mitmproxy
+---------------------------------------------------------------------
+
+(1 row)
+
+SELECT * FROM citus_check_cluster_node_health();
+ from_nodename | from_nodeport | to_nodename | to_nodeport | success
```

Comment:
Hmm, this is actually an unexpected/misleading user experience. The connectivity between the coordinator and 9060 is broken. But, we show `9060` -> `9060` and `9060`-> `57637` connectivity checks as broken. This could lead to very hard times while debugging actual connectivity issues.

We should probably check the return code of `ExecuteOptionalRemoteCommand`. If the return value is NOT `RESPONSE_OKAY`, we should report that connection is also broken. If the local host that we are running is already in PgDistNode, we can actually skip reporting these connectivity checks, because that failure is already going to be reported.

What we can do is something like:
```C

    if response != RESPONSE_OKAY:
            /* we'll going to report this failure anyway, so for these nodes just return NULL indicating that we didn't successfully connect to the source node */
           success = NULL;
          if MyLocalGroupIdInPgDistNode:
                /* we don't need anything special, because this connection pair would be reported anyway*/
          else: 
                add a new row to the output from: LocalHostName GUC / PG knows the current servers, we already know to the node that we fail to connect, false
```

What are your thoughts on this?

Reply:
Another discussion on this can be found here: https://github.com/citusdata/citus/issues/5290#issuecomment-942122862